app/views/unattended/provisioning_templates/snippet/preseed_netplan_generic_interface.erb

```diff
@@ -14,8 +14,8 @@ oses:
 <%- else -%>
     <%= @interface.identifier %>:
 <%- end -%>
-        dhcp4: <%= @dhcp %>
-        dhcp6: <%= @dhcp6 %>
+        dhcp4: <%= !@dhcp.nil? %>
```

Comment:
I would suggest to use `@dhcp.present?` instead.

Reply:
Thanks, fixed!